src/robot/libdocpkg/java9builder.py

```diff
@@ -0,0 +1,150 @@
+                            version=self._get_version(fields),
+                            scope=self._get_scope(fields),
+                            named_args=False,
+                            doc_format=self._get_doc_format(fields))
+        libdoc.inits = self._initializers(elements, constructors)
+        libdoc.keywords = self._keywords(elements, methods)
+        return libdoc
+
+    def _get_doc(self, elements, element):
+        doc_comment = elements.getDocComment(element)
+        if doc_comment is None:
+            return ''
+        else:
+            return '\n'.join(line.strip() for line in doc_comment.splitlines())
+
+    def _get_version(self, fields):
+        return self._get_attr(fields, 'VERSION')
+
+    def _get_scope(self, fields):
+        scope = self._get_attr(fields, 'SCOPE', upper=True)
+        return {'TESTSUITE': 'test suite',
+                'GLOBAL': 'global'}.get(scope, 'test suite')
+
+    def _get_doc_format(self, fields):
+        return self._get_attr(fields, 'DOC_FORMAT', upper=True)
+
+    def _get_attr(self, fields, name, upper=False):
+        name = 'ROBOT_LIBRARY_' + name
+        for field in fields:
+            if field.getSimpleName().toString() == name:
+                value = field.getConstantValue()
+                if upper:
+                    value = utils.normalize(value, ignore='_').upper()
+                return value
+        return ''
+
+    def _initializers(self, elements, constructors):
+        inits = [self._keyword_doc(elements, constructor)
+                 for constructor in constructors]
+        if len(inits) == 1 and not inits[0].args:
+            return []
+        return inits
+
+    def _keywords(self, elements, methods):
+        return [self._keyword_doc(elements, method) for method in methods]
+
+    def _keyword_doc(self, elements, method):
+        doc, tags = utils.split_tags_from_doc(self._get_doc(elements, method))
+        return KeywordDoc(
+            name=utils.printable_name(method.getSimpleName().toString(),
+                                      code_style=True),
+            args=self._get_keyword_arguments(method),
+            doc=doc,
+            tags=tags
+        )
+
+    def _get_keyword_arguments(self, method):
+        params = method.getParameters()
+        if not params:
+            return []
+        names = [param.getSimpleName().toString() for param in params]
+        if self._is_varargs(params[-1]):
+            names[-1] = '*' + names[-1]
+        elif self._is_kwargs(params[-1]):
+            names[-1] = '**' + names[-1]
+            if len(params) > 1 and self._is_varargs(params[-2]):
+                names[-2] = '*' + names[-2]
+        return names
+
+    def _is_varargs(self, param):
+        return (param.asType().toString().startswith('java.util.List') or
+                (param.asType().getKind() == TypeKind.ARRAY and
+                 param.asType().getComponentType().getKind() !=
+                    TypeKind.ARRAY))
```

Comment:
It probably would be a good idea to something like `param_type = param.asType()` in the beginning of this method. Would make other lines simpler and probably the last line would be short enough to fit into single line. Another possibility to handle longish lines cases like this is formatting the code like

```python
return (
    ...
    ...
    ...
)
```